detection_rules/schemas/registry_package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+"""Definitions for packages destined for the registry."""
+
+import dataclasses
+from typing import Dict, Union, Type
+
+import marshmallow_dataclass
+from marshmallow import validate
+
+from .definitions import BaseMarshmallowDataclass, ConditionSemVer, SemVer
+
+
+@marshmallow_dataclass.dataclass
```

Comment:
i don't really like this decorator too much. i think it really mucks with some type hinting from what i've seen

Reply:
I have been on the fence about it too. Biggest difference is it carries the `Schema` within it vs generating with `class_schema`. While it is convenient having the schema embedded, it probably also creates a risk of additional stray properties or attributes.

I'll update to convert to regular `@dataclass` and change how schema is loaded

Reply:
4117d744d2686492ed2c432fe23770abacaa2466